configs/components/curl.rb

```diff
@@ -26,6 +26,7 @@
     pkg.build_requires "runtime-#{settings[:runtime_project]}"
     pkg.environment "PATH" => "$(shell cygpath -u #{settings[:gcc_bindir]}):$(PATH)"
     pkg.environment "CYGWIN" => settings[:cygwin]
+    pkg.environment "NM" => "/bin/nm"
```

Comment:
Can we track this down to the Cygwin change somehow? There's a mirror available at https://github.com/mirror/newlib-cygwin FWIW (but I'm not sure if that's the repo where a change would appear).

I'm interested in how / why this broke (as it may point to other similar things that would break). Also, I never used `nm`, but see that it's similar to `dumpbin` - https://stackoverflow.com/questions/375273/microsoft-equivalent-of-the-nm-command

Reply:
@Iristyle I've done some more digging into this.
The Configure log segments around the checking for the nm tool are interesting:
With the **NM**  fix in place, the configure log is as below:
```
configure:9357: checking command to parse /bin/nm output from x86_64-w64-mingw32-gcc object
configure:9510: x86_64-w64-mingw32-gcc -c  -IC:/tools/pl-build-tools/include -IC:/tools/mingw64/include -IC:/ProgramFiles64Folder/PuppetLabs/Puppet/puppet/include conftest.c >&5
configure:9513: $? = 0
configure:9517: /bin/nm conftest.o \| sed -n -e 's/^.*[ ]\([ABCDGIRSTW][ABCDGIRSTW]*\)[ ][ ]*\([_A-Za-z][_A-Za-z0-9]*\)$/\1 \2 \2/p' | sed '/ __gnu_lto/d' \> conftest.nm
configure:9520: $? = 0
configure:9586: x86_64-w64-mingw32-gcc -o conftest.exe  -IC:/tools/pl-build-tools/include -IC:/tools/mingw64/include -IC:/ProgramFiles64Folder/PuppetLabs/Puppet/puppet/include -LC:/tools/pl-build-tools/lib -LC:/tools/mingw64/lib -LC:/ProgramFiles64Folder/PuppetLabs/Puppet/sys/ruby/lib -Wl,--nxcompat -Wl,--dynamicbase conftest.c conftstm.o >&5
configure:9589: $? = 0
configure:9627: result: ok
```
Whereas without it in place, the log output is:
```
configure:9357: checking command to parse /cygdrive/c/tools/mingw64/bin/nm -B output from gcc object
configure:9510: gcc -c   conftest.c >&5
configure:9513: $? = 0
configure:9517: /cygdrive/c/tools/mingw64/bin/nm -B conftest.o \| sed -n -e 's/^.*[ ]\([ABCDGIRSTW][ABCDGIRSTW]*\)[ ][ ]*\([_A-Za-z][_A-Za-z0-9]*\)$/\1 \2 \2/p' | sed '/ __gnu_lto/d' \> conftest.nm
configure:9520: $? = 0
cannot run sed -n -e 's/^.*[	 ]\([ABCDGIRSTW][ABCDGIRSTW]*\)[	 ][	 ]*\([_A-Za-z][_A-Za-z0-9]*\)$/\1 \2 \2/p' | sed '/ __gnu_lto/d'
configure:9510: gcc -c   conftest.c >&5
configure:9513: $? = 0
configure:9517: /cygdrive/c/tools/mingw64/bin/nm -B conftest.o \| sed -n -e 's/^.*[ ]\([ABCDGIRSTW][ABCDGIRSTW]*\)[ ][ ]*_\([_A-Za-z][_A-Za-z0-9]*\)$/\1 _\2 \2/p' | sed '/ __gnu_lto/d' \> conftest.nm
configure:9520: $? = 0
cannot run sed -n -e 's/^.*[	 ]\([ABCDGIRSTW][ABCDGIRSTW]*\)[	 ][	 ]*_\([_A-Za-z][_A-Za-z0-9]*\)$/\1 _\2 \2/p' | sed '/ __gnu_lto/d'
configure:9624: result: failed
```

Reply:
Further the version of NM being used are:

```
Administrator@u5r3bytw0j6an4l /var/tmp/tmp.MMatSQ2dfp/curl-7.59.0
$ nm --version
GNU nm (GNU Binutils) 2.25
Copyright (C) 2014 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) any later version.
This program has absolutely no warranty.

Administrator@u5r3bytw0j6an4l /var/tmp/tmp.MMatSQ2dfp/curl-7.59.0
$ /bin/nm --version
GNU nm (GNU Binutils) 2.29.1.20171006
Copyright (C) 2017 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) any later version.
This program has absolutely no warranty.
````

Reply:
Version 2.25 (the Mingw version) produces DOS Output, whereas the latest (2.29) adds the LF.
Did a search for configure ```nm``` fails and the only entry I could see was this: http://msys2-users.narkive.com/5jlZFn8i/checking-command-to-parse-nm-output-failed

I have also discovered a slight difference with the ```sed``` command used between the failed/successful run.
So - on the failure (using mingw/nm) - this produces no output:
```
nm ./lib/libcurl_la-base64.o | sed -n -e 's/^.*[ ]\([ABCDGIRSTW][ABCDGIRSTW]*\)[ ][ ]*_\([_A-Za-z][_A-Za-z0-9]*\)$/\1 _\2 \2/p'
[.... no output from this command ...]
```
On the "successful run" using ```/bin/nm``` the ```sed``` command used is:
```
/bin/nm ./lib/libcurl_la-base64.o | sed -n -e 's/^.*[ ]\([ABCDGIRSTW][ABCDGIRSTW]*\)[ ][ ]*\([_A-Za-z][_A-Za-z0-9]*\)$/\1 \2 \2/p'
T Curl_base64_decode Curl_base64_decode
T Curl_base64_encode Curl_base64_encode
T Curl_base64url_encode Curl_base64url_encode
```

The change history for ```nm``` is available at https://sourceware.org/git/gitweb.cgi?p=binutils-gdb.git;a=history;f=binutils/nm.c;h=e46fffc796f772141e9ad8ea4515be2dc36d49cd;hb=HEAD